src/ngraph/runtime/cpu/pass/cpu_layout.cpp

```diff
@@ -990,8 +990,21 @@ namespace ngraph
                     }
                     catch (const mkldnn::error& e)
                     {
-                        throw ngraph_error("MKLDNN Unsupported pooling layout" +
-                                           to_string(input_desc.data.format) + e.message);
+                        if (arg0_shape.size() == 4 || arg0_shape.size() == 5)
+                        {
+                            auto default_format = arg0_shape.size() == 4
+                                                      ? mkldnn::memory::format::nchw
+                                                      : mkldnn::memory::format::ncdhw;
+                            auto default_desc = mkldnn_utils::create_default_mkldnn_md(
+                                node.get(), 0, false, default_format);
+                            i_mds.push_back(default_desc);
+                            o_mds.push_back(default_desc);
```

Comment:
Should default_desc be mkldnn_utils::create_default_mkldnn_md(node.get(), 0, true, default_format) for o_mds? Or you just let the next node to take care of it?

Reply:
I guess here it make sense, since mkldnn keeps the result in the same layout as that of input.

Reply:
The output shape is not always the same as the input shape for pooling.